pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -797,13 +797,6 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 				log.Warnf("hydrateWorkloadInstance for %v failed: %v", proxy.ID, err)
 			}
 		} else if pod != nil {
-			// for split horizon EDS k8s multi cluster, in case there are pods of the same ip across clusters,
-			// which can happen when multi clusters using same pod cidr.
-			// As we have proxy Network meta, compare it with the network which endpoint belongs to,
-			// if they are not same, ignore the pod, because the pod is in another cluster.
-			if proxy.Metadata.Network != c.endpointNetwork(proxyIP) {
```

Comment:
Would it make sense to just duplicate the same cluster ID check here that we do in the aggregate controller (just for correctness)?

Reply:
Yeah, good idea. Added that.

Reply:
Still want to keep it within the branch at least - I think proxies attached to VMs/WLEs are intended to use this code, although I'm not sure how they ever would with the check in the aggregate controller.